receipt/src/test/java/com/hyperwallet/android/receipt/repository/ReceiptDataSourceTest.java

```diff
@@ -0,0 +1,384 @@
+
+        // error occurred, this will save params and callback
+        assertThat(mReceiptDataSource.getErrors().getValue(), is(notNullValue()));
+
+        // test retry, saved params and callback will be used and no null pointer exception is thrown
+        mReceiptDataSource.retry();
+
+        // verify calls
+        verify(mReceiptDataSource, times(2)).loadInitial(
+                ArgumentMatchers.<PageKeyedDataSource.LoadInitialParams<Integer>>any(),
+                ArgumentMatchers.<PageKeyedDataSource.LoadInitialCallback<Integer, Receipt>>any());
+        verify(mReceiptDataSource, never()).loadAfter(
+                ArgumentMatchers.<PageKeyedDataSource.LoadParams<Integer>>any(),
+                ArgumentMatchers.<PageKeyedDataSource.LoadCallback<Integer, Receipt>>any());
+    }
+
+    @Test
+    public void testLoadAfter_returnsReceipts() throws Exception {
+        String json = mExternalResourceManager.getResourceContent("receipt_list_date_grouping_response.json");
+        JSONObject jsonObject = new JSONObject(json);
+        final HyperwalletPageList<Receipt> response = new HyperwalletPageList<>(jsonObject, Receipt.class);
+        doAnswer(new Answer() {
+            @Override
+            public Object answer(InvocationOnMock invocation) {
+                HyperwalletListener listener = (HyperwalletListener) invocation.getArguments()[1];
+                listener.onSuccess(response);
+                return listener;
+            }
+        }).when(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+
+        // test
+        mReceiptDataSource.loadAfter(mLoadAfterParams, mLoadAfterCallback);
+
+        verify(mHyperwallet).listReceipts(any(ReceiptQueryParam.class),
+                ArgumentMatchers.<HyperwalletListener<HyperwalletPageList<Receipt>>>any());
+        verify(mLoadAfterCallback).onResult(mListArgumentCaptor.capture(), mNextCaptor.capture());
+
+        assertThat(mNextCaptor.getValue(), is(10));
+
+        // assert receipts information
+        List<Receipt> receipts = mListArgumentCaptor.getValue();
+        assertThat(receipts, Matchers.<Receipt>hasSize(5));
+        assertThat(receipts.get(3).getJournalId(), is("51660675"));
+        assertThat(receipts.get(3).getType(), is(PAYMENT));
+        assertThat(receipts.get(3).getEntry(), is(CREDIT));
+        assertThat(receipts.get(3).getSourceToken(), is("act-b1f6dc28-e534-45f4-a661-3523f051f77a"));
+        assertThat(receipts.get(3).getDestinationToken(), is("usr-b4e8ec34-52d8-4a81-9566-bdde1bd745b6"));
+        assertThat(receipts.get(3).getAmount(), is("13.00"));
+        assertThat(receipts.get(3).getFee(), is("0.00"));
+        assertThat(receipts.get(3).getCurrency(), is("USD"));
+        assertThat(receipts.get(3).getDetails(), is(notNullValue()));
+        assertThat(receipts.get(3).getDetails().getPayeeName(), is("kevin Puckett"));
```

Comment:
just being a little picky here, use upper case in Kevin. there are more cases like that too